extensions/amp-share-tracking/0.1/test/test-amp-share-tracking.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS-IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createIframePromise} from '../../../../testing/iframe';
+import {AmpShareTracking} from '../amp-share-tracking';
+import {Viewer} from '../../../../src/service/viewer-impl';
+import {Xhr} from '../../../../src/service/xhr-impl';
+import {toggleExperiment} from '../../../../src/experiments';
+import * as sinon from 'sinon';
+
+describe('amp-share-tracking', () => {
+  let sandbox;
+  let viewerForMock;
+  let xhrMock;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    viewerForMock = sandbox.stub(Viewer.prototype, 'getFragment');
+    xhrMock = sandbox.stub(Xhr.prototype, 'fetchJson');
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  function getAmpShareTracking(optVendorUrl) {
+    return createIframePromise().then(iframe => {
+      toggleExperiment(iframe.win, 'amp-share-tracking', true);
+      const el = iframe.doc.createElement('amp-share-tracking');
+      if (optVendorUrl) {
+        el.setAttribute('data-href', optVendorUrl);
+      }
+      const ampShareTracking = new AmpShareTracking(el);
+      return ampShareTracking;
+    });
+  }
+
+  it('should get incoming fragment starting with dot', () => {
+    viewerForMock.onFirstCall().returns(Promise.resolve('.12345'));
+    const mockJsonResponse = {fragment: '54321'};
+    xhrMock.onFirstCall().returns(Promise.resolve(mockJsonResponse));
+    return getAmpShareTracking('http://foo.bar').then(ampShareTracking => {
+      ampShareTracking.buildCallback();
+      expect(ampShareTracking.vendorHref_).to.equal('http://foo.bar');
```

Comment:
`vendorHref_` is an private member field, instead of checking its value in the test, a better way is to verify how it is used. In this case, you can do:
`xhrMock.withArgs('http://foo.bar')`